examples/megacli.cpp

```diff
@@ -1931,62 +1943,68 @@ int loadfile(string* name, string* data)
     return 0;
 }
 
-void xferq(direction_t d, int cancel)
+void xferq(direction_t d, int cancel, bool showActive, bool showAll, bool showCount)
 {
     string name;
+    int count = 0, activeCount = 0;
 
     DBTableTransactionCommitter committer(client->tctable);
     for (appfile_list::iterator it = appxferq[d].begin(); it != appxferq[d].end(); )
     {
         if (cancel < 0 || cancel == (*it)->seqno)
         {
+            bool active = (*it)->transfer && (*it)->transfer->slot;
             (*it)->displayname(&name);
 
-            cout << (*it)->seqno << ": " << name;
-
-            if (d == PUT)
+            if (active && showActive || showAll)
             {
-                AppFilePut* f = (AppFilePut*) *it;
+                cout << (*it)->seqno << ": " << name;
```

Comment:
We should not print directly into the standard output channel, but in the log. I think we should adjust these `cout`s to print into a stream and later on print the log message (despite the `cout` were already there). What do you think?

Reply:
This function xferq() is called from both exec_putq and exec_getq which are invoked by the megacli user in order to see the state of their transfer queue, those have been in megacli forever.  I think in this case we should continue to output to the console so the user can see it when they request it, though there are plenty of other cases where we output probably too much stuff that the user may not want to see, that could be logged instead.  thanks

Reply:
Oh, you are right. I thought it was inside the SDK, not at `megacli` level. Otherwise, I agree, the `cout` was already there and there are plenty of them spread along the code of the example. Nevermind